src/components/Common/UpdatableApp.tsx

```diff
@@ -37,147 +31,14 @@ export const checkForUpdate = async () => {
   const meta = await res.json();
 
   if (appVersion !== meta.version) {
-    // Trigger an update if key: 'app-version' not present in localStorage
-    // or does not match with metaVersion.
     console.info("App can be updated.");
     return meta.version as string;
   }
 };
 
 const UpdatableApp = ({ children, silentlyAutoUpdate }: UpdatableAppProps) => {
-  const [newVersion, setNewVersion] = useState<string>();
-  const [appUpdated, setAppUpdated] = useState(false);
-
-  useEffect(() => {
-    checkForUpdate()
-      .then(setNewVersion)
-      .then(() => {
-        const appUpdated = localStorage.getItem(APP_UPDATED_KEY);
-        if (appUpdated === "true") {
-          setAppUpdated(true);
-        }
-      });
-  }, []);
-
-  useEffect(() => {
-    if (!appUpdated) return;
-
-    localStorage.removeItem(APP_UPDATED_KEY);
-    setTimeout(() => setAppUpdated(false), 5000);
-  }, [appUpdated]);
-
-  const updateApp = async () => {
-    if (!newVersion) return;
-
-    // Service worker cache should be cleared with caches.delete()
-    caches.keys().then((names) => names.forEach((name) => caches.delete(name)));
-
-    // A second of delay to appreciate the update animation.
-    const updateLocalStorageAndReload = () => {
-      localStorage.setItem(APP_UPDATED_KEY, "true");
-      window.location.reload();
-      localStorage.setItem(APP_VERSION_KEY, newVersion);
-    };
-
-    silentlyAutoUpdate
-      ? updateLocalStorageAndReload()
-      : setTimeout(updateLocalStorageAndReload, 1000);
-  };
-
-  if (newVersion && silentlyAutoUpdate) updateApp();
-
-  return (
-    <div className="relative">
-      {children}
-      {newVersion && <UpdateAppPopup onUpdate={updateApp} />}
-      <AppUpdatedAlert show={appUpdated} />
-    </div>
-  );
+  useAppUpdates(silentlyAutoUpdate);
+  return <div className="relative">{children}</div>;
```

Comment:
is this needed anymore? 🤔 absolute positioning was needed for headless ui's popover alone right?

```suggestion
  return children;
```

Reply:
I’ll check if any child components still rely on relative. If not, we can remove it.